pkg/core/dns/cache.go

```diff
@@ -0,0 +1,39 @@
+package dns
```

Comment:
nit: Should the package be `dns_cache`?

Reply:
I think Nikolay is kind of right here. I'd like to avoid confusion that this is part of DNS server. How about `dns/lookup` package with `lookup.go` (or `interface.go`) and `cache.go` files?

Reply:
Fixed, create package `lookup` in `dns` dir and 2 files: `cache.go` and `interface.go`